docs/source/include/learning.rst

```diff
@@ -0,0 +1,93 @@
+
+
+Unrolled Optimization Algorithms
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Different iterative solution algorithms produce different ODP networks. The ones implemented in SCICO are
+
+* Proximal Map: It corresponds to solving
+
+    .. math::
+       :label: eq:odp_prox
+
+       \argmin_{\mb{x}} \; \alpha_k \, f(A \mb{x}, \mb{y}) + \frac{1}{2} \| \mb{x} - \mb{x}^k - \mb{x}^{k+1/2} \|_2^2 \;,
+
+with :math:`k` corresponding to the index of the iteration, which translates to an index of the stage of the network, :math:`f(A \mb{x}, \mb{y})` a fidelity term, usually an :math:`\ell_2` norm, and :math:`\mb{x}^{k+1/2}` a regularization representing :math:`\mathrm{prox}_r (\mb{x}^k)` and usually implemented as a convolutional neural network CNN. This proximal map representation is used when the minimization problem in :eq:`eq:odp_prox` can be solved as a relatively simple analytical computation.
+
+:class:`.ODPProxDnBlock` uses this formulation to solve a denoising problem, yielding
+
+    .. math::
+       \mb{x}^{k+1} = (\alpha_k \, \mb{y} + \mb{x}^k + \mb{x}^{k+1/2}) \, / \, (\alpha_k + 1) \;,
+
+where :math:`A` corresponds to the identity operator and is therefore omitted, :math:`\mb{y}` is the noisy signal, :math:`\alpha_k > 0` is a learned stage-wise parameter weighting the contribution of the fidelity term and :math:`\mb{x}^k + \mb{x}^{k+1/2}` is the regularization, usually represented by a residual CNN.
+
+
+:class:`.ODPProxDblrBlock` uses this formulation to solve a deblurring problem
+
+    .. math::
+       \mb{x}^{k+1} = \mathcal{F}^{-1} \mathrm{diag} (\alpha_k | \mathcal{K}|^2 + 1 )^{-1} \mathcal{F} \, (\alpha_k K^T * \mb{y} + \mb{x}^k + \mb{x}^{k+1/2}) \;,
+
+    where :math:`A` is the blurring operator, :math:`K` is the blurring kernel, :math:`\mb{y}` is the blurred signal, :math:`\mathcal{F}` is the DFT, :math:`\mathcal{K}` is the DFT of :math:`K`, :math:`\alpha_k > 0` is a learned  stage-wise parameter weighting the contribution of the fidelity term and :math:`\mb{x}^k + \mb{x}^{k+1/2}` is the regularization represented by a residual CNN.
+
+
+* Gradient Descent: When the solution of the optimization problem in :eq:`eq:odp_prox` can not be simply represented by an analytical step, a formulation based on a gradient descent iteration is preferred. This yields
+
+    .. math::
+       \mb{x}^{k+1} = \mb{x}^k + \mb{x}^{k+1/2} - \alpha_k \, A^T \nabla_x \, f(A \mb{x}^k, \mb{y}) \;,
+
+where :math:`\mb{x}^{k+1/2}` represents :math:`\nabla r(\mb{x}^k)`.
+
+:class:`.ODPGrDescBlock` uses this formulation to solve a generic problem with :math:`\ell_2` fidelity as
+
+    .. math::
+       \mb{x}^{k+1} = \mb{x}^k + \mb{x}^{k+1/2} - \alpha_k \, A^T (A \mb{x} - \mb{y}) \;,
+
+with :math:`\mb{y}` the measured signal and :math:`\mb{x} + \mb{x}^{k+1/2}` a residual CNN net.
+
```

Comment:
`net` at the end of `CNN net` is redundant.

Reply:
Removed.